account_invoice_payment_report/models/account_invoice.py

```diff
@@ -0,0 +1,108 @@
+from odoo.tools import float_is_zero
+
+
+class AccountInvoice(models.Model):
+
+    _inherit = 'account.invoice'
+
+    # Backport of odoo v12.0 method
+    def _get_payments_vals(self):
+
+        self.ensure_one()
+        if not self.payment_move_line_ids:
+            return []
+        payment_vals = []
+        currency_id = self.currency_id
+        for payment in self.payment_move_line_ids:
+            payment_currency_id = False
+            if self.type in ('out_invoice', 'in_refund'):
+                amount = sum(
+                    [
+                        p.amount
+                        for p in payment.matched_debit_ids
+                        if p.debit_move_id in self.move_id.line_ids
+                    ]
+                )
+                amount_currency = sum(
+                    [
+                        p.amount_currency
+                        for p in payment.matched_debit_ids
+                        if p.debit_move_id in self.move_id.line_ids
+                    ]
+                )
+                if payment.matched_debit_ids:
+                    payment_currency_id = (
+                        all(
+                            [
+                                p.currency_id
+                                == payment.matched_debit_ids[0].currency_id
+                                for p in payment.matched_debit_ids
+                            ]
+                        )
+                        and payment.matched_debit_ids[0].currency_id
+                        or False
+                    )
+            elif self.type in ('in_invoice', 'out_refund'):
+                amount = sum(
+                    [
+                        p.amount
+                        for p in payment.matched_credit_ids
+                        if p.credit_move_id in self.move_id.line_ids
+                    ]
+                )
+                amount_currency = sum(
+                    [
+                        p.amount_currency
+                        for p in payment.matched_credit_ids
+                        if p.credit_move_id in self.move_id.line_ids
+                    ]
+                )
+                if payment.matched_credit_ids:
+                    payment_currency_id = (
+                        all(
+                            [
+                                p.currency_id
+                                == payment.matched_credit_ids[0].currency_id
+                                for p in payment.matched_credit_ids
+                            ]
+                        )
+                        and payment.matched_credit_ids[0].currency_id
+                        or False
+                    )
+            # get the payment value in invoice currency
+            if payment_currency_id and payment_currency_id == self.currency_id:
+                amount_to_show = amount_currency
+            else:
+                currency = payment.company_id.currency_id
```

Comment:
If you rename the variable `currency_id` as proposed, you need to have a different variable name here.
s/currency/company_currency